src/kv/change_set.h

```diff
@@ -2,20 +2,31 @@
 // Licensed under the Apache 2.0 License.
 #pragma once
 
-#include "ds/champ_map.h"
 #include "ds/hash.h"
 #include "kv/kv_types.h"
+#ifndef KV_STATE_RB
+#  include "ds/champ_map.h"
+#else
+#  include "ds/rb_map.h"
+#endif
 
 #include <map>
 
 namespace kv
 {
   template <typename V>
   using VersionV = map::VersionV<V>;
+#ifndef KV_STATE_RB
   template <typename K, typename V, typename H>
   using State = champ::Map<K, VersionV<V>, H>;
   template <typename K, typename V, typename H>
   using Snapshot = champ::Snapshot<K, VersionV<V>, H>;
+#else
+  template <typename K, typename V, typename H>
+  using State = rb::Map<K, VersionV<V>>;
```

Comment:
fwiw, you could save a few lines with single block where you do `using State = CCF_MAP_NS::Map<K, VersionV<V>>;` etc and a single `#define CCF_MAP_NS rb|champ" at lines 8 and 9 where you already have a the switch.

Reply:
I thought about this too but because the `champ::Map` takes an extra template parameter `H` (that the `rb::Map` doesn't), it's a little awkward. I've simplified it as much as I think is possible though.

Reply:
Oops, indeed, I'd missed that!